### PR TITLE
fix #5250: append_by_permutation bug for BinaryColumn

### LINKED ISSUE #5250: The query returned wrong result after set `sql_select_limit = 9999999`.
### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...'
```
            CREATE TABLE `test_sql_select_limit`(
                c1 int,
                c2 varchar(20))
                ENGINE=OLAP
                DUPLICATE KEY(`c1`)
                COMMENT "OLAP"
                DISTRIBUTED BY HASH(`c1`) BUCKETS 5
                PROPERTIES (
                "replication_num" = "1",
                "in_memory" = "false",
                "storage_format" = "DEFAULT"
                );
```
2. INSERT INTO '....'
```
insert into test_sql_select_limit values(1992, 'luoyonghao'),(2008, 'peace and love');
```
3. SELECT  '....'
```
mysql> set sql_select_limit = 99;
Query OK, 0 rows affected (0.02 sec)

mysql> select * from test_sql_select_limit order by 1 desc;
+------+----------------+
| c1   | c2             |
+------+----------------+
| 2008 | peace and love |
| 1992 | luoyonghao     |
+------+----------------+
2 rows in set (0.00 sec)

mysql> set sql_select_limit = 9999999;
Query OK, 0 rows affected (0.00 sec)

mysql> select * from test_sql_select_limit order by 1 desc;
+------+----------------+
| c1   | c2             |
+------+----------------+
| 2008 | `WX�   tionBl  |
| 1992 | �8_�     f       |
+------+----------------+
2 rows in set (0.01 sec)
```
### Expected behavior (Required)
```
mysql> select * from test_sql_select_limit order by 1 desc;
+------+----------------+
| c1   | c2             |
+------+----------------+
| 2008 | peace and love |
| 1992 | luoyonghao     |
+------+----------------+
2 rows in set (0.00 sec)
```
### Real behavior (Required)
```
mysql> select * from test_sql_select_limit order by 1 desc;
+------+----------------+
| c1   | c2             |
+------+----------------+
| 2008 | `WX�   tionBl  |
| 1992 | �8_�     f       |
+------+----------------+
2 rows in set (0.01 sec)
```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 - e12dea3

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #5250 

## Problem Summary(Required) ：

Offset calculation bug in `append_by_permutation` for BinaryColumn.

### CODE REVIEW

File: be/src/exec/vectorized/sorting/sort_permute.cpp [hunk 1]
```diff
@@ -156,7 +156,7 @@ class ColumnAppendPermutation final : public ColumnVisitorMutableAdapter<ColumnA
 
         auto& offsets = dst->get_offset();
         auto& bytes = dst->get_bytes();
-        size_t old_rows = offsets.size();
+        size_t old_rows = offsets.size() - 1;
         size_t total_rows = offsets.size();
```
Comment (trueeyu) [inline be/src/exec/vectorized/sorting/sort_permute.cpp:160]: why don't change `total rows `

Comment (trueeyu) [reply be/src/exec/vectorized/sorting/sort_permute.cpp:160]: The semantics of total_rows and old_rows are different， may be misunderstood

File: be/src/exec/vectorized/sorting/sort_permute.cpp [hunk 2]
```diff
@@ -156,7 +156,7 @@ class ColumnAppendPermutation final : public ColumnVisitorMutableAdapter<ColumnA
 
         auto& offsets = dst->get_offset();
         auto& bytes = dst->get_bytes();
-        size_t old_rows = offsets.size();
+        size_t old_rows = offsets.size() - 1;
```
[the remark this answers is not in the document]
Comment (mofeiatwork) [reply be/src/exec/vectorized/sorting/sort_permute.cpp:159]: done

File: be/src/exec/vectorized/sorting/sort_permute.cpp (hunk 1 above)
Comment (mofeiatwork) [reply be/src/exec/vectorized/sorting/sort_permute.cpp]: rename it to `num_offsets`, it's number of offsets, not rows.

**Review by trueeyu — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**